pkg/apis/storage/validation/validation.go

```diff
@@ -404,12 +404,11 @@ func ValidateCSIDriver(csiDriver *storage.CSIDriver) field.ErrorList {
 // ValidateCSIDriverUpdate validates a CSIDriver.
 func ValidateCSIDriverUpdate(new, old *storage.CSIDriver) field.ErrorList {
 	allErrs := ValidateCSIDriver(new)
-
-	// Spec is read-only
-	// If this ever relaxes in the future, make sure to increment the Generation number in PrepareForUpdate
-	if !apiequality.Semantic.DeepEqual(old.Spec, new.Spec) {
-		allErrs = append(allErrs, field.Invalid(field.NewPath("spec"), new.Spec, "field is immutable"))
-	}
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(old.Spec.AttachRequired, new.Spec.AttachRequired, field.NewPath("spec", "attachRequired"))...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(old.Spec.FSGroupPolicy, new.Spec.FSGroupPolicy, field.NewPath("spec", "fsGroupPolicy"))...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(old.Spec.VolumeLifecycleModes, new.Spec.VolumeLifecycleModes, field.NewPath("spec", "volumeLifecycleModes"))...)
+	allErrs = append(allErrs, apivalidation.ValidateImmutableField(old.Spec.StorageCapacity, new.Spec.StorageCapacity, field.NewPath("spec", "storageCapacity"))...)
```

Comment:
just curious, why do need to validate only this 4 fields, not the others from `Spec`?

Reply:
The way that we've validated immutability for other objects when only some of the fields are mutable:
1. we validate the mutable fields first
2. then we set it back to the original value
3. then check that the spec is immutable

that way we won't accidentally miss validating some fieldsgg